### PR TITLE
Fix saml configuration

### PR DESCRIPTION
**Pull request by ethervoid**
Fixes https://github.com/CartoDB/support/issues/1167
JSON serialized returns the Hash with string instead of symbol in the keys and this default/nil values when try to get them

### CODE REVIEW

File: lib/carto/saml_service.rb [hunk 1]
```diff
@@ -125,7 +125,8 @@ def saml_settings(settings_hash = carto_saml_configuration)
     end
 
     def carto_saml_configuration
-      @organization.try(:auth_saml_configuration)
+      saml_config = @organization.try(:auth_saml_configuration)
```
Comment (juanignaciosl) [inline lib/carto/saml_service.rb]: Suggestion: `@organization.try(:auth_saml_configuration).try(:deep_symbolize_keys)`.

File: spec/lib/carto/saml_service_spec.rb [hunk 2]
```diff
@@ -26,6 +26,10 @@
     it 'is enabled if there is configuration' do
       service.enabled?.should be_true
     end
+
+    it 'email_attribute not the default if defined' do
```
Comment (juanignaciosl) [inline spec/lib/carto/saml_service_spec.rb]: In this test, it's not obvious how come the return value should be `username`, and it depends on the factory value. Please consider adding within the test a explicit assignment of a value to `email_attribute`. It'd also be nice setting it to `nil` and testing the default value.


Minor: I think that "email_attribute not the default if defined" should be written as "email_attribute doesn't return the default value if it's defined".

**Comment by ramiroaznar** [comment]
@juanignaciosl because Mario is off, maybe you can appy this merge, right?

**Comment by AleGoiko** [comment]
hey @ramiroaznar I think it's on me to do the acceptance, will try to do this asap :)

**Comment by juanignaciosl** [comment]
@AleGoiko thank you, let me know if I can help!

**Pull request merged**